albumentations/augmentations/functional.py

```diff
@@ -1256,3 +1256,80 @@ def py3round(number):
 
 def noop(input_obj, **params):
     return input_obj
+
+
+def split_and_shuffle_shape_by_grid(shape, grid, random_state=42):
+    """
+    Split image by grid.
+
+    Args:
+        shape (tuple): image shape.
+        grid (tuple): tuple with two values.
+        random_state (int): seed for numpy
+    """
+    np.random.seed(random_state)
```

Comment:
we don't modify random seed in functions. we instantiate all params in class. please see how it's done in other classes